pkg/webhook/admission/pod/storage_initializer_injector.go

```diff
@@ -330,12 +331,26 @@ func (mi *StorageInitializerInjector) InjectStorageInitializer(pod *v1.Pod) erro
 		}
 	}
 
-	// Allow to override the uid for the case where ISTIO CNI with DNS proxy is enabled
-	// See for more: https://istio.io/latest/docs/setup/additional-setup/cni/#compatibility-with-application-init-containers.
-	if value, ok := pod.GetAnnotations()[constants.IstioSidecarUIDAnnotationKey]; ok {
-		if uid, err := strconv.ParseInt(value, 10, 64); err == nil {
-			initContainer.SecurityContext.RunAsUser = ptr.Int64(uid)
-		}
+	/*
+		OpenShift uses istio-cni which causes an issue with init-containers when calling external services
+		like S3 or similar. Setting the `uid` for the `storage-initializer` to the same `uid` as the
+		`uid` of the `istio-proxy` resolves the issue. In OpenShift the `istio-proxy` always gets assigned
+		the first `uid` from the namespaces `uid` range + 1 (The range is defined in an annotation on the namespace).
+	*/
+	if initContainer.SecurityContext == nil {
```

Comment:
Wouldn't it be better to move this functionality in a dedicated initializer instead of putting it into ta "storage initializer" ? Semantically I don't think its a good fit here to mangle with the security context. It also should be easier to add/remove an initializer at an higher level instead of modifying an initializer (and from the outside its not clear that a storage related initializer also manages the security context)

Reply:
The change is only relevant for the `storage-initializer` and these are the bits of code that create the `init-container` spec for that. I agree with you semantically, but having it somewhere else would mean that we have to know how to filter the `storage-initializer` and modifying its security context there. I find it more clear here.

Reply:
Just saw that this is coming from upstream and shouldn't be changed here but in upstream kserve. So forget this comment for now.